File: kotlin-spring/src/main/kotlin/org/openapitools/model/ResponseTimeFilterProperties.kt
package org.openapitools.model

import java.util.Objects
import com.fasterxml.jackson.annotation.JsonProperty
import org.openapitools.model.ResponseDistanceBreakdownItem
import org.openapitools.model.ResponseFares
import org.openapitools.model.ResponseRoute
import javax.validation.constraints.DecimalMax
import javax.validation.constraints.DecimalMin
import javax.validation.constraints.Max
import javax.validation.constraints.Min
import javax.validation.constraints.NotNull
import javax.validation.constraints.Pattern
import javax.validation.constraints.Size
import javax.validation.Valid

/**
 * 
 * @param travelTime 
 * @param distance 
 * @param distanceBreakdown 
 * @param fares 
 * @param route 
 */
data class ResponseTimeFilterProperties(

    @field:JsonProperty("travel_time") val travelTime: kotlin.Int? = null,

    @field:JsonProperty("distance") val distance: kotlin.Int? = null,

    @field:Valid
    @field:JsonProperty("distance_breakdown") val distanceBreakdown: kotlin.collections.List<ResponseDistanceBreakdownItem>? = null,

    @field:Valid
    @field:JsonProperty("fares") val fares: ResponseFares? = null,

    @field:Valid
    @field:JsonProperty("route") val route: ResponseRoute? = null
) {

}

File: scala-akka-http-server/src/main/scala/org/openapitools/server/model/ResponseTimeFilterProperties.scala
package org.openapitools.server.model


/**
 * @param travelTime  for example: ''null''
 * @param distance  for example: ''null''
 * @param distanceBreakdown  for example: ''null''
 * @param fares  for example: ''null''
 * @param route  for example: ''null''
*/
final case class ResponseTimeFilterProperties (
  travelTime: Option[Int],
  distance: Option[Int],
  distanceBreakdown: Option[Seq[ResponseDistanceBreakdownItem]],
  fares: Option[ResponseFares],
  route: Option[ResponseRoute]
)

